src/commonMain/kotlin/komem/litmus/runners/LitmusTestRunnerUtil.kt

```diff
@@ -0,0 +1,74 @@
+package komem.litmus.runners
+
+import komem.litmus.*
+import kotlin.time.Duration
+import kotlin.time.TimeSource
+
+fun LitmusTestRunner.runTest(
+    timeLimit: Duration,
+    params: RunningParams,
+    testProducer: () -> LitmusTest,
+): LitmusResult {
+    val results = mutableListOf<LitmusOutcomeInfo>()
+    val start = TimeSource.Monotonic.markNow()
+    while (start.elapsedNow() < timeLimit) {
+        val outcomes = runTest(params, testProducer)
+        val outcomeSetup = testProducer().getOutcomeSetup()
+        results.addAll(outcomes.groupToInfo(outcomeSetup))
+    }
+    return results.mergeOutcomes()
+}
+
+/*
+ Note: interprets AffinityMap as a sequence of smaller maps
+ Example: for a map [ [0], [1], [2], [3] ],a test with 2 threads, and 2 instances, the
+ first instance will have a [ [0], [1] ] map and the second one will have [ [2], [3] ].
+ */
+fun LitmusTestRunner.runTestParallel(
+    params: RunningParams,
+    instances: Int,
+    testProducer: () -> LitmusTest,
+): List<LitmusOutcome> {
+    val threadCount = testProducer().overriddenThreads().size
+    val allOutcomes = List(instances) { instanceIndex ->
+        val newAffinityMap = params.affinityMap?.let { oldMap ->
+            AffinityMap { threadIndex ->
+                oldMap.allowedCores(instanceIndex * threadCount + threadIndex)
+            }
+        }
+        val newParams = params.copy(affinityMap = newAffinityMap)
+        runTest(newParams, testProducer)
+    }
+    return allOutcomes.flatten()
+}
+
+fun LitmusTestRunner.runTestParallel(
+    params: RunningParams,
+    testProducer: () -> LitmusTest,
+): List<LitmusOutcome> = with(cpuCount() / testProducer().overriddenThreads().size) {
```

Comment:
`with` is overused here (you use it for `Int`).
`with` is generally used to call a bunch a functions in a context of some object:
https://kotlinlang.org/docs/scope-functions.html#with

Usually there is no point to use `with` with primitive types. 

Besides, the code without `with` is much cleaner here. 

```
val instances = cpuCount() / testProducer().overriddenThreads().size
return runTestParallel(params, instances, testProducer)
```

or 


```
... = runTestParallel(params = params, testProducer = testProducer, instances = cpuCount() / testProducer().overriddenThreads().size)
```

Reply:
I saved one line though! Fine, fixed